dotnet/src/SemanticKernel.Abstractions/SemanticFunctions/PromptConfig.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.TemplateEngine;
+using Microsoft.SemanticKernel.Text;
+using static Microsoft.SemanticKernel.SemanticFunctions.PromptTemplateConfig;
+
+namespace Microsoft.SemanticKernel.SemanticFunctions;
+
+/// <summary>
+/// Prompt configuration.
+/// </summary>
+public class PromptConfig
+{
+    /// <summary>
+    /// Input parameter for semantic functions.
+    /// </summary>
+    public class InputParameter
+    {
+        /// <summary>
+        /// Name of the parameter to pass to the function.
+        /// e.g. when using "{{$input}}" the name is "input", when using "{{$style}}" the name is "style", etc.
+        /// </summary>
+        [JsonPropertyName("name")]
+        [JsonPropertyOrder(1)]
+        public string Name { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Parameter description for UI apps and planner. Localization is not supported here.
+        /// </summary>
+        [JsonPropertyName("description")]
+        [JsonPropertyOrder(2)]
+        public string Description { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Default value when nothing is provided.
+        /// </summary>
+        [JsonPropertyName("defaultValue")]
+        [JsonPropertyOrder(3)]
+        public string DefaultValue { get; set; } = string.Empty;
+    }
+
+    /// <summary>
+    /// Settings for a prompt request. This supports text completions, image generation, etc.
+    /// </summary>
+    public class PromptRequestSettings
+    {
+        /// <summary>
+        /// Order for this model.
+        /// </summary>
+        public int? order { get; set; }
```

Comment:
Hi @SergeyMenshykh 
I'll fix the capitalisation. Order won't be the only option used to determine which model is selected but supporting different strategies to select the model is not part of this PR it will come later.